src/components/implementation/cbuf_mgr/default/cbuf_mgr.c

```diff
@@ -643,8 +648,21 @@ cbuf_create(spdid_t spdid, unsigned long size, int cbid)
 		cbi->owner.spdid = spdid;
 		INIT_LIST(&cbi->owner, next, prev);
 		INIT_LIST(cbi, next, prev);
+	
+		if (cbi_head) {
+			struct cbuf_info *cbi_new = (struct cbuf_info *) malloc(sizeof(struct cbuf_info));
+			cbi_new->cbid = id;
+
+			ADD_END_LIST(cbi_head, cbi_new, next, prev);
+		}
+		else {
+			cbi_head = (struct cbuf_info *) malloc(sizeof(struct cbuf_info));
+			cbi_head->cbid = id;
+			
+			INIT_LIST(cbi_head, next, prev);
+		}
```

Comment:
You can simplify this code a bit by moving the first two lines out of the `if`.
Here you allocate another `struct cbuf_info` for the same cbuf, so be careful to maintain both structures.
Or you can add another set of `next/prev` pointer to the `struct cbuf_info`.
But if you want to change this global list approach in the future, it is fine now.